core/src/main/scala/chisel3/properties/Object.scala

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.properties
+
+import chisel3.SpecifiedDirection
+import chisel3.internal.{throwException, HasId, NamedComponent, ObjectFieldBinding}
+
+import scala.collection.immutable.HashMap
+
+/** Represents an instance of a Class.
+  *
+  * This cannot be instantiated directly, instead see Class.unsafeGetDynamicObject.
+  *
+  * The DynamicObject is generally unsafe, in that its getField method does not check the name, type, or direction of
+  * the accessed field. It may be used with care, and a more typesafe version will be added.
+  */
+case class DynamicObject private[chisel3] (className: String) extends HasId with NamedComponent {
```

Comment:
```suggestion
class DynamicObject private[chisel3] (className: String) extends HasId with NamedComponent {
```
case class implies certain things that `HadId` does not follow.